### ISSUE TITLE
How to use xonsh --shell-type=readline on Windows?

### ISSUE THREAD

**Issue by halloleo** [xonsh/xonsh#3409]
I have a brand-new install of Python with xonsh on Windows 7. I need to use `--shell-type=readline`, so I am wondering which readline library I should install.

I tried `pyreadline` 2.1, but with that I get the depreciation warning `C:\sw\Python\Python38\lib\site-packages\pyreadline\py3k_compat.py:8: Deprecation Warning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated`. This warning is known from other third-party modules in xonsh (see e.g.  issue #2872).

*Note:* `prompt-toolkit` is not an option because it doesn't work with `--shell-type=readline`.

**Comment by gforsyth** [comment]
Hey @halloleo -- @melund may be able to weight in here, but I know we've strongly recommended using `prompt_toolkit` on windows for a while now because the various readline options were suboptimal.

Is there a particular reason you need to use the readline shell on Windows?

**Comment by halloleo** [comment]
Yep, indeed: I am running xonsh inside an Emacs shell-buffer and for this xonsh needs to interface via simple readline - because [prompt_toolkit  cannot degrade to a dump termina](https://github.com/prompt-toolkit/python-prompt-toolkit/issues/1032).

**Comment by jonathanslenders** [comment]
I'm working on a fix for this: https://github.com/prompt-toolkit/python-prompt-toolkit/pull/1035

**Comment by scopatz** [comment]
readline doesn't really work on windows (outside of the WLS)

**Comment by anki-code** [comment]
This looks solved and should be closed.

**Comment by halloleo** [comment]
No, no, cannot be closed yet. It needs to be tested, whether ptk is a viable alternative to readline on dump terminals! I will try it when I get around to it.

**Comment by anki-code** [comment]
I'm going to close this till new reports about testing.
